src/mapgen/mapgen.cpp

```diff
@@ -1063,9 +1063,20 @@ void MapgenParams::writeParams(Settings *settings) const
 }
 
 
-// Calculate exact edges of the outermost mapchunks that are within the
-// set 'mapgen_limit'.
-void MapgenParams::calcMapgenEdges()
+s32 MapgenParams::getSpawnRangeMax()
+{
+	if (!m_mapgen_edges_calculated) {
+		std::pair<s16, s16> edges = get_mapgen_edges(mapgen_limit, chunksize);
+		mapgen_edge_min = edges.first;
```

Comment:
Why can't it be cached in `MapgenParams::calcMapgenEdges()` like before? `ModApiMapgen` does have accesss to the function via `getServer(L)->getEmergeManager()->map_settings_mgr->mapgen_params`.

Reply:
If the function is called while mods are still loading, it will have to call `makeMapgenParams` to get the parameters. After `makeMapgenParams` is called, other mods will no longer be able to change mapgen settings: https://github.com/minetest/minetest/blob/38169db765f2553316ae3f1c0f40b0f7e9154b49/src/map_settings_manager.cpp#L67-L71

Reply:
Fair point. Would you mind adding a comment about that? Someone might want to "simplify" or "optimize" it.